core-bundle/src/Controller/SitemapController.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Controller;
+
+use Contao\Backend;
+use Contao\CoreBundle\Event\ContaoCoreEvents;
+use Contao\CoreBundle\Event\SitemapEvent;
+use Contao\PageModel;
+use Contao\System;
+use Contao\User;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * @Route(defaults={"_scope" = "frontend"})
+ *
+ * @internal
+ */
+class SitemapController extends AbstractController
+{
+    /**
+     * @Route("/sitemap.xml")
+     */
+    public function __invoke(Request $request): Response
+    {
+        $this->initializeContaoFramework();
+
+        /** @var PageModel $pageModel */
+        $pageModel = $this->get('contao.framework')->getAdapter(PageModel::class);
+        $rootPages = $pageModel->findPublishedRootPages(['dns' => $request->server->get('HTTP_HOST')]);
+
+        if (null === $rootPages) {
+            return new Response('', Response::HTTP_NOT_FOUND);
+        }
+
+        $pages = [];
+        $rootPageIds = [];
+        $tags = ['contao.sitemap'];
+
+        foreach ($rootPages as $rootPage) {
+            /** @var Backend $backend */
+            $backend = $this->get('contao.framework')->getAdapter(Backend::class);
+            $pages = array_merge($pages, $backend->findSearchablePages($rootPage->id, '', true));
+            $pages = $this->callLegacyHook($rootPage, $pages);
```

Comment:
Is it intended that this call is made in each iteration with the same growing array? (`$pages`)

Reply:
Yes, that's the current behaviour of `getSearchablePages()`.